metecho/api/models.py

```diff
@@ -1591,6 +1611,25 @@ def finalize_commit_dataset(self, *, error=None, originating_user_id):
                 originating_user_id=originating_user_id,
             )
 
+    def finalize_commit_omnistudio(self, *, error=None, originating_user_id):
+        self.currently_retrieving_omnistudio = False
+        self.save()
+        if error is None:
+            self.notify_changed(
+                type_="SCRATCH_ORG_COMMIT_OMNISTUDIO",
+                originating_user_id=originating_user_id,
+            )
+            if self.task:
+                self.task.finalize_commit_changes(
+                    originating_user_id=originating_user_id
+                )
+        else:
+            self.notify_scratch_org_error(
+                error=error,
+                type_="SCRATCH_ORG_COMMIT_OMNISTUDIO_FAILED",
+                originating_user_id=originating_user_id,
+            )
```

Comment:
💭 : Feels like [the underlying abstraction](https://github.com/SFDO-Tooling/Metecho/blob/c9eac68fa088b892bb304ad1c8e6eb2300cff971/metecho/api/push.py#L107) could be improved, but this is fine as-is.